sky/jobs/utils.py

```diff
@@ -270,22 +270,28 @@ def _handle_legacy_job(job_id: int):
             failure_reason = ('Inconsistent internal job state. This is a bug.')
         elif pid is None:
             # Non-legacy job and controller process has not yet started.
-            controller_status = job_lib.get_status(job_id)
-            if controller_status in [
-                    job_lib.JobStatus.FAILED_SETUP,
-                    job_lib.JobStatus.FAILED_DRIVER, job_lib.JobStatus.FAILED
-            ]:
-                # We should fail the case where the controller status is
-                # failed, as it is likely due to the job for submitting the
-                # managed job to scheduler failed.
-                logger.error('Failed to submit the managed job to scheduler.')
-            elif schedule_state in (
-                    managed_job_state.ManagedJobScheduleState.INACTIVE,
-                    managed_job_state.ManagedJobScheduleState.WAITING):
+            if (schedule_state ==
+                    managed_job_state.ManagedJobScheduleState.INACTIVE):
+                controller_status = job_lib.get_status(job_id)
+                if controller_status in [
+                        job_lib.JobStatus.FAILED_SETUP,
+                        job_lib.JobStatus.FAILED_DRIVER,
+                        job_lib.JobStatus.FAILED
+                ]:
+                    # We should fail the case where the controller status is
```

Comment:
we need to double check the schedule_state after job_lib.get_status to avoid a race

Reply:
Hmm, good point! I tried to add the schedule state check again here, and the logic becomes quite complicated. I now revert it to only check the `FAILED_SETUP` case, which should be safe to only check the job_lib state. PTAL.

Reply:
Yeah, I think it is okay now.